internal/cli/plugin/plugin_github_asset_manager.go

```diff
@@ -43,55 +42,63 @@ var (
-type GithubRelease struct {
-	owner   string
-	name    string
-	version *semver.Version
+type GithubAssetManager struct {
+	ghClient *github.Client
+	owner    string
+	name     string
+	version  *semver.Version
 }
 
-type AssetOpts struct {
-	existingCommands []*cobra.Command
-	ghClient         *github.Client
-	githubRelease    *GithubRelease
+func (g *GithubAssetManager) repository() string {
+	return fmt.Sprintf("%s/%s", g.owner, g.name)
 }
 
-func (opts *AssetOpts) repository() string {
-	if opts.githubRelease == nil {
-		return ""
+func createGithubAssetManagerFromPlugin(p *plugin.Plugin) (*GithubAssetManager, error) {
+	if !p.HasGithub() {
+		return nil, errCreatePluginAssetFromPlugin
 	}
-	return fmt.Sprintf("%s/%s", opts.githubRelease.owner, opts.githubRelease.name)
+
+	return &GithubAssetManager{
+		owner: p.Github.Owner,
+		name:  p.Github.Name,
+	}, nil
+}
+
+func (g *GithubAssetManager) getPluginDirectoryName() string {
+	return fmt.Sprintf("%s@%s", g.owner, g.name)
 }
 
-func (opts *AssetOpts) getPluginAssetInfo() ([]*github.ReleaseAsset, error) {
+func (g *GithubAssetManager) getPluginAssetInfo() ([]*github.ReleaseAsset, error) {
 	var err error
 	var release *github.RepositoryRelease
 
 	// download latest release if version is not specified
-	if opts.githubRelease.version == nil {
-		release, _, err = opts.ghClient.Repositories.GetLatestRelease(context.Background(), opts.githubRelease.owner, opts.githubRelease.name)
+	if g.version == nil {
+		release, _, err = g.ghClient.Repositories.GetLatestRelease(context.Background(), g.owner, g.name)
 
 		if err != nil {
-			return nil, fmt.Errorf("could not find latest release for %s", opts.repository())
+			return nil, fmt.Errorf("could not find latest release for %s", g.repository())
 		}
 	} else {
 		// try to find the release with the version tag with v prefix, if it does not exist try again without the prefix
-		release, _, err = opts.ghClient.Repositories.GetReleaseByTag(context.Background(), opts.githubRelease.owner, opts.githubRelease.name, "v"+opts.githubRelease.version.String())
+		release, _, err = g.ghClient.Repositories.GetReleaseByTag(context.Background(), g.owner, g.name, "v"+g.version.String())
 
 		if release == nil || err != nil {
-			release, _, err = opts.ghClient.Repositories.GetReleaseByTag(context.Background(), opts.githubRelease.owner, opts.githubRelease.name, opts.githubRelease.version.String())
+			release, _, err = g.ghClient.Repositories.GetReleaseByTag(context.Background(), g.owner, g.name, g.version.String())
 		}
 
 		if err != nil {
-			return nil, fmt.Errorf("could not find the release %s release for %s", opts.githubRelease.name, opts.repository())
+			return nil, fmt.Errorf("could not find the release %s release for %s", g.name, g.repository())
 		}
 	}
 
 	return release.Assets, nil
 }
 
-func (opts *AssetOpts) getAssetID(assets []*github.ReleaseAsset) (int64, error) {
+func (g *GithubAssetManager) getAssetID(assets []*github.ReleaseAsset) (int64, error) {
```

Comment:
Would it make sense to just call this `getID`?
Since the name of this struct will already have the word asset in it? (`GithubAsset`)

Reply:
done